ocaml/squeezed/lib/squeeze.ml

```diff
@@ -807,18 +800,21 @@ let balance_memory ?fistpoints io =
 (** Return true if the host memory is currently unbalanced and needs rebalancing *)
 let is_host_memory_unbalanced ?fistpoints io =
   let debug_string, host = io.make_host () in
-  let domains = List.map (fun d -> (d.domid, d)) host.domains in
   let _, _, _, result =
     Squeezer.one_iteration ?fistpoints false (is_balanced io) (Squeezer.make ())
       host io.target_host_free_mem_kib (io.gettimeofday ())
   in
   let is_new_target a =
-    let existing = (List.assoc a.action_domid domains).target_kib in
-    a.new_target_kib <> existing
+    let existing_target_kib =
+      host.domains
+      |> DomainSet.find_first_opt (fun d -> a.action_domid = d.domid)
```

Comment:
This doesn't work as you think it does:: the function given to find_first_opt must be monotonic, that is, at some point all elements _must_ return true. This means `find_first_opt` is not useful for equality.

One working alternative would be to create and empty domain, with `action_domid` as it's `domid`, and then use `find_opt` on this new element